src/cli/subcommands/sync_cmd.rs

```diff
@@ -231,3 +236,25 @@ async fn wait_for_snapshot_completion(
 ) -> anyhow::Result<SnapshotProgressState> {
     check_snapshot_progress(client, true).await
 }
+
+/// Handles the initial check for snapshot download if the node is initializing.
+async fn handle_initial_snapshot_check(client: &rpc::Client) -> anyhow::Result<()> {
+    let initial_report = SyncStatus::call(client, ()).await?;
+    // Use the public getter method instead of accessing the private field
```

Comment:
this comment is not clear to me, like which private field and why ?

Reply:
Added the comment as a reminder to use `getters` instead of directly accessing private fields during refactoring, but forgot to remove it.

Reply:
Removed the comment.